core/PRP/Geometry/plGBufferGroup.cpp

```diff
@@ -547,6 +560,113 @@ void plGBufferGroup::addIndices(const std::vector<unsigned short>& indices)
         fIdxBuffStorage[idx][i] = indices[i];
 }
 
+void plGBufferGroup::IPackIcicle(const plGeometrySpan* geoSpan, plIcicle* ice)
+{
+    size_t i;
+    for (i = 0; i < fIdxBuffStorage.size(); ++i)
+        if (geoSpan->getNumIndices() + fIdxBuffCounts[i] < kMaxNumIndicesPerBuffer)
+            break;
+    if (i == fIdxBuffStorage.size()) {
+        fIdxBuffStorage.push_back(nullptr);
+        fIdxBuffCounts.push_back(0);
+    }
+
+    unsigned int idxOffset = fIdxBuffCounts[i];
+    unsigned short* idxStorage = new unsigned short[idxOffset + geoSpan->getNumIndices()];
+    if (idxOffset != 0)
+        memcpy(idxStorage, fIdxBuffStorage[i], idxOffset * sizeof(unsigned short));
+    fIdxBuffCounts[i] += geoSpan->getNumIndices();
+    delete fIdxBuffStorage[i];
+    fIdxBuffStorage[i] = idxStorage;
+
+    for (size_t j = 0; j < geoSpan->getNumIndices(); ++j)
+        idxStorage[idxOffset + j] = geoSpan->getIndex(j) + (unsigned short)ice->getVStartIdx();
+
+    ice->setIBufferIdx(i);
+    ice->setIStartIdx(idxOffset);
+    ice->setILength(geoSpan->getNumIndices());
+}
+
+void plGBufferGroup::IPackVertexSpan(const plGeometrySpan* geoSpan, plVertexSpan* vSpan)
+{
+    size_t i;
+    for (i = 0; i < fVertBuffStorage.size(); ++i)
+        if (geoSpan->getNumVertices() + getNumVertices(i) < kMaxNumVertsPerBuffer)
+            break;
```

Comment:
Please use braces around multiline loop bodies, even if they are not syntactically necessary.